tests/type_inference_test.py

```diff
@@ -0,0 +1,193 @@
+import unittest
+import numpy as np
+from dace.config import Config
+from dace.codegen.tools import type_inference
+from dace import dtypes
+import ast
+
+
+class TestTypeInference(unittest.TestCase):
+    def testSimpleAssignement(self):
+        # simple assignment tests
+
+        #bool
+        code_str = "value=True"
+        inf_symbols = type_inference.infer(code_str, {})
```

Comment:
perhaps infer should have a default argument for symbols (`=None`, **not** `={}`) just to make things cleaner.